state/tx/indexer/batch.go

```diff
@@ -0,0 +1,32 @@
+package indexer
+
+import "github.com/tendermint/tendermint/types"
+
+// A Batch groups together multiple Index operations you would like performed
+// at the same time. The Batch structure is NOT thread-safe. You should only
+// perform operations on a batch from a single thread at a time. Once batch
+// execution has started, you may not modify it.
+type Batch struct {
+	Ops map[string]types.TxResult
+}
+
+// NewBatch creates a new Batch.
+func NewBatch() *Batch {
+	return &Batch{
+		Ops: make(map[string]types.TxResult),
+	}
+}
+
+// Index adds or updates entry for the given hash.
+func (b *Batch) Index(hash []byte, result types.TxResult) error {
```

Comment:
But we're not setting anything... Maybe `Add` if you don't like the word index. Also, `Tx` seems redundant since we already have the subject in the package name (`txindexer`). Don't you think?